### ISSUE TITLE
Fatal Error: Call to a member function get() on null in object-cache.php

### ISSUE THREAD

**Issue by kingkool68** [stuttter/ludicrousdb#62]
Installed LudicriousDB via Composer in the `mu-plugins` directory, moved the drop-in files into place, and setup my configuration. When I go to look at my test site I see a PHP Fatal Error:

```
Fatal error: Uncaught Error: Call to a member function get() on null in /var/www/spiritedmedia.dev/htdocs/wp-content/object-cache.php:129 Stack trace: #0 /var/www/spiritedmedia.dev/htdocs/wp-content/mu-plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php(1931): wp_cache_get('localhost:3306', 'ludicrousdb') #1 /var/www/spiritedmedia.dev/htdocs/wp-content/mu-plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php(1662): LudicrousDB->tcp_cache_get('localhost:3306') #2 /var/www/spiritedmedia.dev/htdocs/wp-content/mu-plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php(638): LudicrousDB->check_tcp_responsiveness('localhost', 3306, 0.2) #3 /var/www/spiritedmedia.dev/htdocs/wp-content/mu-plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php(1266): LudicrousDB->db_connect('SELECT meta_val...') #4 /var/www/spiritedmedia.dev/htdocs/wp-includes/wp-db.php(2488): LudicrousDB->query('SELECT meta_val...') #5 /var/www/spiritedmedia.dev/htdocs/wp-content/object-cache.php(1196): wpdb->get_results('SELECT in /var/www/spiritedmedia.dev/htdocs/wp-content/object-cache.php on line 129
```

We're using the latest version of [WP Redis](https://wordpress.org/plugins/wp-redis/) for our object caching. 

It looks like the global `$wp_object_cache` isn't set, aka its `null` when LudicrousDB tries to get something from the cache.

**Comment by kkmuffme** [comment]
Tested with other cache plugins, works perfectly fine. Issue seems to be with WP Redis plugin

**Comment by spacedmonkey** [comment]
This seems to be an issue with WP Redis. So I am going to close this ticket out

**Comment by partounian** [comment]
Anyone happen to have a solution or better redis plugin alternative?

**Comment by partounian** [comment]
Fixed by updated wp_cache_init to

```
    global $wp_object_cache;

    if (! ($wp_object_cache instanceof WP_Object_Cache)) {
        $wp_object_cache = new WP_Object_Cache;
    }

```

**Comment by spacedmonkey** [comment]
Reopenning as I believe I have a solution. In https://github.com/stuttter/ludicrousdb/pull/67 I check if object cache is enabled before writing to it. This fixes other race conditions

**Comment by spacedmonkey** [comment]
For those interested, this should be fixed in 4.2. Let me know if you have this issue again.
